captum/attr/_core/layer/layer_ablation.py

```diff
@@ -0,0 +1,273 @@
+            >>> # The attributions can be calculated as follows:
+            >>> # layer mask has dimensions 1 x 3 x 3
+            >>> layer_mask = torch.tensor([[[0,0,0],[1,1,1],
+            >>>                             [2,2,2]]])
+            >>> attr = ablator.attribute(input, target=1,
+            >>>                          layer_mask=layer_mask)
+        """
+
+        def layer_forward_func(*args):
+            layer_input = args[0]
+            original_inputs = args[1:]
+
+            device_ids = self.device_ids
+            if (
+                device_ids is None
+                and isinstance(self.forward_func, torch.nn.DataParallel)
+                and self.forward_func.device_ids is not None
+            ):
+                device_ids = self.forward_func.device_ids
+
+            all_layer_inputs = {}
+            if device_ids is not None:
+                scattered_layer_input, _ = scatter_kwargs(
+                    (layer_input,), None, target_gpus=device_ids
+                )
+                for tensor in scattered_layer_input:
+                    all_layer_inputs[tensor[0].device] = tensor[0]
+            else:
+                all_layer_inputs[layer_input.device] = layer_input
+
+            with torch.no_grad():
+
+                def forward_hook(module, inp, out=None):
+                    if inp[0].device not in all_layer_inputs:
+                        raise AssertionError(
+                            "Layer input not placed on appropriate "
+                            "device. If using a DataParallel model, either provide the "
+                            "DataParallel model as forward_func or provide device ids"
+                            " to the constructor."
+                        )
+                    return all_layer_inputs[inp[0].device]
+
+                if attribute_to_layer_input:
+                    hook = self.layer.register_forward_pre_hook(forward_hook)
+                else:
+                    hook = self.layer.register_forward_hook(forward_hook)
+                eval = _run_forward(self.forward_func, original_inputs, target=target)
+                hook.remove()
+                return eval
+
+        inputs = _format_input(inputs)
+        additional_forward_args = _format_additional_forward_args(
+            additional_forward_args
+        )
+        layer_eval = _forward_layer_eval(
+            self.forward_func,
+            inputs,
+            self.layer,
+            additional_forward_args,
+            device_ids=self.device_ids,
+            attribute_to_layer_input=attribute_to_layer_input,
+        )
+        all_inputs = (
+            (inputs + additional_forward_args)
+            if additional_forward_args is not None
+            else inputs
+        )
+
+        ablator = FeatureAblation(layer_forward_func)
+
+        return ablator.attribute(
+            layer_eval,
+            baselines=baselines,
```

Comment:
Makes sense, yeah, it's a different here from baselines in other algorithms. I think since we want to evaluate essentially "turning off" each layer input / output, it's not clear what the layer values would be for a given input baseline. Maybe we can document the difference between this baseline and others more clearly? Or rename the argument to something like layer_baseline instead?